### PR TITLE
Add test to ensure existence of component docs

### PR DESCRIPTION
**Pull request by asuresh4**
**Description:** Add a simple go test to ensure existence of documentation for enabled components.

**Link to tracking Issue:** #1025

### CODE REVIEW

**Comment by asuresh4** [comment]
FYI @tigrannajaryan @jrcamp

**Comment by jrcamp** [comment]
I have a couple suggestions on possible simplifications, easier to show in a diff: https://github.com/jrcamp/opentelemetry-collector/commit/2863ce36946c8c7883707e3523de38f8593488a9

Ideally we can reuse this in contrib but I think some of the values will need to be parameterized. Are you planning to do those changes later or should we make it usable for both?

**Comment by asuresh4** [comment]
Thanks @jrcamp for the suggestions. Will incorporate those into this PR. As a first step I was planning on adding a similar test to the contrib repo. When you say reusable, where would the test live?

**Comment by jrcamp** [comment]
> Thanks @jrcamp for the suggestions. Will incorporate those into this PR. As a first step I was planning on adding a similar test to the contrib repo. When you say reusable, where would the test live?

I think the test would live in core but it would have to be in a non-test package. Maybe see `componenttest` package as an example. There would have to be an actual test function in each but it can just call the shared package.

File: component/componenttest/component_docs.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+var componentTypes = []string{"extension", "receiver", "processor", "exporter"}
+
+// getImportPrefixesToCheck returns a slice of strings that are relevant import
+// prefixes for components in the given module.
+func getImportPrefixesToCheck(module string) []string {
+	out := make([]string, len(componentTypes))
+	for i, typ := range componentTypes {
+		out[i] = fmt.Sprintf("%s/%s", strings.TrimRight(module, "/"), typ)
+	}
+	return out
```
Comment (jrcamp) [inline component/componenttest/component_docs.go:91]: Can you use `strings.Join`?

**Comment by jrcamp** [comment]
lgtm, just one minor improvement suggested above. Looks like coverage isn't high enough yet though.

**Comment by asuresh4** [comment]
> lgtm, just one minor improvement suggested above. Looks like coverage isn't high enough yet though.

Addressed your comment. Still trying to get codecov/patch happy though.

File: component/componentdocs/component_docs.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+//		"service/defaultcomponents/defaults.go",
+//      "go.opentelemetry.io/collector",
+//	)
+//
+// 2) Usage in the contrib repo:
+// componenttest.VerifyComponentDocumentation(
+//		"path/to/project",
+//		"cmd/otelcontrib/components.go",
+//      "github.com/open-telemetry/opentelemetry-collector-contrib",
+//	).
```
Comment (tigrannajaryan) [inline component/componentdocs/component_docs.go:51]: Nice comment!

File: component/componentdocs/component_docs_test.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	type args struct {
+		projectPath                   string
+		relativeDefaultComponentsPath string
+		projectGoModule               string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
```
Comment (tigrannajaryan) [inline component/componentdocs/component_docs_test.go:113]: Good to see all these cases tested, thanks.

**Pull request merged**